### DISCUSSION TITLE
no gles support for gtk host

### DISCUSSION THREAD

**Discussion by dougcooper** [unoplatform/uno#9694] [General]
The software appears to have the hooks in place for gles support, however:

* the gtk interface [pulls in](https://github.com/unoplatform/uno/blob/00843186a0a4be8750530a41a50d12524b83bc2f/src/Uno.UI.Runtime.Skia.Gtk/OpenGLESRenderSurface.cs#L30) the GL libraries not the GLES libraries
* we modified the uno lib to pull in the expected libs but still received an error: `glDrawBuffer() not found` which means something in the lower layers is trying to make a call that is implementation dependent.

The [opengl](https://docs.gl/gl3/glDrawBuffer) docs do confirm that this is an unsupported call for GLES.

**Comment by jeromelaban** [comment]
Thanks. The code you're mentioning does import the OpenGL ES portions (the `using` at the top indicates that), though we're still having issues with the autodetection of the appropriate OpenGL flavor to use.

Try using this to explicitly use the ES flavor:
```csharp
var host = new GtkHost(() => new App());
host.RenderSurfaceType = RenderSurfaceType.OpenGLES;
```

**Reply by dougcooper** [reply]
im not sure what `using` statement your referring to. the `OpenGLESRenderSurface` constructor explicitly uses `GLCoreLibraryNameContainer` instead of `OpenGLESLibraryNameContainer` which brings in the non-gles lib versions. After making that change we also had to ensure `GDK_GL=gles` was set and symlink to a non-versioned shared object and it detected just fine.

**Reply by jeromelaban** [reply]
Indeed, you're right, the resolved that is incorrect. Interestingly, we did not detect this in our validations, meaning that something was changing the outcome. Are you testing on Linux directly?

**Reply by dougcooper** [reply]
yes...directly on an embedded device

**Reply by jeromelaban** [reply]
Thanks. We'll take a look. It also looks similar to https://github.com/unoplatform/uno/issues/9099.

**Reply by dougcooper** [reply]
A little backstory...the app was running fine on the embedded device except for some stuttering during a small animation as compared to a non-embedded version of the app. We initially tried forcing RenderSurfaceType as you suggested which caused the app to fail except for the Software type.
